storage/src/main/java/io/atomix/storage/journal/SegmentedJournal.java

```diff
@@ -467,8 +475,10 @@ synchronized void removeSegment(JournalSegment segment) {
    * @param descriptor The segment descriptor.
    * @return The segment instance.
    */
-  protected JournalSegment<E> newSegment(JournalSegmentFile segmentFile, JournalSegmentDescriptor descriptor) {
-    return new JournalSegment<>(segmentFile, descriptor, storageLevel, maxEntrySize, indexDensity, namespace);
+  protected JournalSegment<E> newSegment(JournalSegmentFile segmentFile,
+      JournalSegmentDescriptor descriptor) {
+    return new JournalSegment<>(segmentFile, descriptor, storageLevel, maxEntrySize, namespace,
+        journalIndexFactory.get());
```

Comment:
Does it make sense to get a new factory every time when getting a new segment? Why not just have the journal factory in the constructor and use the same journal index always?